hail_search/hail_search_query.py

```diff
@@ -625,6 +654,63 @@ def _format_transcript_args(self):
         })
         return args
 
+    def _annotate_allowed_consequences(self, annotations, annotation_filters):
+        allowed_consequences = {
+            ann for field, anns in (annotations or {}).items()
+            if anns and (field not in ANNOTATION_OVERRIDE_FIELDS) for ann in anns
+        }
+        consequence_enum = self._get_enum_lookup('sorted_transcript_consequences', 'consequence_term')
+        allowed_consequence_ids = {consequence_enum[c] for c in allowed_consequences if consequence_enum.get(c)}
+
+        annotation_exprs = {}
+        if allowed_consequence_ids:
+            allowed_consequence_ids = hl.set(allowed_consequence_ids)
+            allowed_transcripts = self._ht.sorted_transcript_consequences.filter(
+                lambda tc: tc.consequence_term_ids.any(allowed_consequence_ids.contains)
+            )
+            annotation_exprs['allowed_transcripts'] = allowed_transcripts
+            annotation_filters = annotation_filters + [hl.is_defined(allowed_transcripts.first())]
+
+        consequence_filter = self._or_filter(annotation_filters)
+        if consequence_filter is not None:
+            annotation_exprs['has_allowed_annotation'] = consequence_filter
+
+        if annotation_exprs:
+            self._ht = self._ht.annotate(**annotation_exprs)
+
+    def _get_annotation_override_filters(self, pathogenicity, annotations):
+        annotation_filters = []
+
+        for key, *args in self.PATHOGENICITY_FILTERS:
+            path_terms = (pathogenicity or {}).get(key)
+            if path_terms:
+                annotation_filters.append(self._has_terms_range_expr(path_terms, key, *args))
+        if (annotations or {}).get(SCREEN_KEY):
+            screen_enum = self._get_enum_lookup(SCREEN_KEY.lower(), 'region_type')
+            allowed_consequences = hl.set({screen_enum[c] for c in annotations[SCREEN_KEY]})
+            annotation_filters.append(allowed_consequences.contains(self._ht.screen.region_type_ids.first()))
+        if (annotations or {}).get(SPLICE_AI_FIELD):
```

Comment:
It may be cleaner to update the `annotations` with `annotations = annotations or {}` before using it.